File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/Battery/BatteryActionsTableViewController.m
//
//  BatteryActionsTableViewController.m
//  DJISdkDemo
//
//  Copyright © 2015 DJI. All rights reserved.
//

#import <DJISDK/DJISDK.h>
#import "DemoUtility.h"
#import "BatteryActionsTableViewController.h"
#import "BatterySelfDischargeViewController.h"

@implementation BatteryActionsTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.sectionNames = [NSMutableArray arrayWithArray:@[@"General"]];
    [self.items addObject:@[[DemoSettingItem itemWithName:@"Set/Get Self-discharge Day" andClass:[BatterySelfDischargeViewController class]]]];
}

-(DJIBaseComponent *)getComponent {
    return [DemoComponentHelper fetchBattery]; 
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Components/Battery/BatteryActionsTableViewController.swift
//
//  BatteryActionsTableViewController.swift
//  DJISDKSwiftDemo
//
//  Created by DJI on 12/17/15.
//  Copyright © 2015 DJI. All rights reserved.
//

import DJISDK

class BatteryActionsTableViewController: DemoTableViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        self.connectedComponent = ConnectedProductManager.sharedInstance.fetchBattery()
        self.showComponentVersionSn = true
        self.sectionNames=["Battery"]
        self.items.append(DemoSettingItem(name: "Set/Get Self-discharge Day", andClass:BatterySelfDischargeViewController.self))
    }    
}
